pdns/dnsdistdist/doh.cc

```diff
@@ -333,6 +333,16 @@ try
   h2o_socket_getsockname(sock, reinterpret_cast<struct sockaddr*>(&local));
   DOHServerConfig* dsc = reinterpret_cast<DOHServerConfig*>(req->conn->ctx->storage.entries[0].data);
 
+  // I wanted to put this near the other h2o_set_header method call, but should I move it?
+  for (auto const& headerPair : dsc->df->d_customResponseHeaders) {
+    const h2o_token_t *token = h2o_lookup_token(headerPair.first.c_str(), headerPair.first.size());
+    // TODO: Should I provide so WARN logging when we can't find a matching token?
+    if (token != NULL) {
+      // h2o_set_header(pool, headers, token, value, value_len, overwrite_if_exists)
+      h2o_set_header(&req->pool, &req->res.headers, token, headerPair.second.c_str(), headerPair.second.size(), 1);
```

Comment:
How about calling `h2o_set_header_by_str()` instead? It will do the lookup for you (provided that you set `maybe_token` to 1) so it should be as fast, but will also add an unknown (one for which there is no token) header.

Reply:
Do we want to allow unknown headers?  I thought this was a surety concern as well as a performance consideration.

Reply:
I think we should allow it, if it really has a negative impact the admin can just stick to known ones :)